### PR TITLE
indices create tool

### PR DESCRIPTION
**Pull request by miiu96**
- Tool that will create indices mappings with a custom config

### CODE REVIEW

File: tools/indexes-creator/cmd/indexes-creator/main.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+   {{.Version}}
+   {{end}}
+`
+
+func main() {
+	app := cli.NewApp()
+	cli.AppHelpTemplate = helpTemplate
+	app.Name = "Index cr"
+	app.Version = "v1.0.0"
+	app.Usage = "This is the entry point for Elasticsearch reindexing tool"
```
Comment (bogdan-rosianu) [inline tools/indexes-creator/cmd/indexes-creator/main.go:62]: `Elasticsearch indices creator tool` *

rename `indexes` to `indices` for directory name, structs, and so on

File: tools/indexes-creator/cmd/indexes-creator/main.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	"github.com/ElrondNetwork/elastic-indexer-go/tools/indexes-creator/reader"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/pelletier/go-toml"
+	"github.com/urfave/cli"
+)
+
+const configFileName = "cluster.toml"
+
+type Config struct {
```
Comment (bogdan-rosianu) [inline tools/indexes-creator/cmd/indexes-creator/main.go:20]: make this struct unexported

File: tools/indexes-creator/cmd/indexes-creator/config/cluster.toml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[config]
+    url = "http://localhost:9200"
```
Comment (bogdan-rosianu) [inline tools/indexes-creator/cmd/indexes-creator/config/cluster.toml:2]: align all items in regards to whitespaces

File: tools/indexes-creator/cmd/indexes-creator/main.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+	}
+
+	pathToMappings := cfgPath + "noKibana"
+	if cfg.ClusterConfig.UseKibana {
+		pathToMappings = cfgPath + "withKibana"
+	}
+
+	indexesMappings, _, err := reader.GetElasticTemplatesAndPolicies(pathToMappings, cfg.ClusterConfig.EnabledIndexes)
+	if err != nil {
+		log.Error("cannot load templates", "error", err.Error())
```
Comment (bogdan-rosianu) [inline tools/indexes-creator/cmd/indexes-creator/main.go:99]: mirght return if error

File: tools/indexes-creator/cmd/indexes-creator/main.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+func createIndexesAndMappings(ctx *cli.Context) {
+	cfgPath := ctx.String(configPath.Name)
+	cfg, err := loadConfigFile(cfgPath)
+	if err != nil {
+		log.Error("cannot load config file", "error", err.Error())
+	}
+
+	pathToMappings := cfgPath + "noKibana"
```
Comment (bogdan-rosianu) [inline tools/indexes-creator/cmd/indexes-creator/main.go:92]: please set the default value of the config directory to `"./config"` and here use `fmt.Sprintf("%s/%s, cfgPath, "noKibana")`

File: tools/indexes-creator/cmd/indexes-creator/main.go [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+	}
+
+	indexesMappings, _, err := reader.GetElasticTemplatesAndPolicies(pathToMappings, cfg.ClusterConfig.EnabledIndexes)
+	if err != nil {
+		log.Error("cannot load templates", "error", err.Error())
+	}
+
+	err = createTemplates(cfg, indexesMappings)
+	if err != nil {
+		log.Error("cannot create templates", "error", err.Error())
```
Comment (bogdan-rosianu) [inline tools/indexes-creator/cmd/indexes-creator/main.go:104]: might return if error

File: tools/indexes-creator/cmd/indexes-creator/main.go [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+	}
+
+	log.Info("All indexes were created")
```
Comment (bogdan-rosianu) [inline tools/indexes-creator/cmd/indexes-creator/main.go:107]: all*

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: tools/indexes-creator/cmd/indexes-creator/main.go (hunk 1 above)
Comment (miiu96) [reply tools/indexes-creator/cmd/indexes-creator/main.go:62]: done.

File: tools/indexes-creator/cmd/indexes-creator/main.go (hunk 2 above)
Comment (miiu96) [reply tools/indexes-creator/cmd/indexes-creator/main.go:20]: done.

File: tools/indexes-creator/cmd/indexes-creator/config/cluster.toml (hunk 3 above)
Comment (miiu96) [reply tools/indexes-creator/cmd/indexes-creator/config/cluster.toml:2]: done.

File: tools/indexes-creator/cmd/indexes-creator/main.go (hunk 4 above)
Comment (miiu96) [reply tools/indexes-creator/cmd/indexes-creator/main.go:99]: done.

File: tools/indexes-creator/cmd/indexes-creator/main.go (hunk 5 above)
Comment (miiu96) [reply tools/indexes-creator/cmd/indexes-creator/main.go:92]: done.

File: tools/indexes-creator/cmd/indexes-creator/main.go (hunk 7 above)
Comment (miiu96) [reply tools/indexes-creator/cmd/indexes-creator/main.go:107]: done.

**Review by dragos-rebegea — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**